addons/udes_sale_stock/tests/test_full_sale_reservation.py

```diff
@@ -0,0 +1,194 @@
+"""Unit tests for full sale reservation"""
+from unittest import mock
+from odoo.addons.udes_common.tests.common import SavepointMixin
+from .common import BaseSaleUDES
+import datetime
+from odoo.exceptions import UserError
+
+
+class TestFullSaleReservation(BaseSaleUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestFullSaleReservation, cls).setUpClass()
+
+        Location = cls.env["stock.location"]
+        cls.Move = cls.env["stock.move"]
+        cls.Picking = cls.env["stock.picking"]
+
+        picking_zone = Location.create(
+            {
+                "name": "Picking Zone",
+                "barcode": "LPICKINGZONE",
+                "usage": "view",
+                "location_id": cls.warehouse.id,
+            }
+        )
+        cls.pick_location = Location.create(
+            {
+                "name": "Pick Location 01",
+                "barcode": "LPICKLOCATION01",
+                "usage": "internal",
+                "location_id": picking_zone.id,
+            }
+        )
+        picking_zone2 = Location.create(
+            {
+                "name": "Picking Zone 2",
+                "barcode": "LPICKINGZONE2",
+                "usage": "view",
+                "location_id": cls.warehouse.id,
+            }
+        )
+        cls.pick_location_2 = Location.create(
+            {
+                "name": "Pick Location 02",
+                "barcode": "LPICKLOCATION02",
+                "usage": "internal",
+                "location_id": picking_zone2.id,
+            }
+        )
+        cls.picking_type_pick.default_location_src_id = picking_zone
+        cls.picking_type_pick2 = cls.picking_type_pick.copy({"default_location_src_id": picking_zone2.id})
+
+        products_info = [{"product": cls.apple, "uom_qty": 10.0}]
+
+        cls.picking_type_pick.u_num_reservable_pickings = -1
```

Comment:
nit: if picking_type_pick2 (the copy) has all this config the same, we could .copy() after setting it on picking_type_pick to avoid needing to set it all twice